packages/shared/src/variants/baduk_utils.ts

```diff
@@ -67,22 +68,22 @@ export type BoardShape = "1d" | "2d" | "flatten-2d-to-1d";
 export function mapBoard<T, S>(board: T[], f: (x: T) => S, shape: "1d"): S[];
 export function mapBoard<T, S>(
   board: T[][],
-  f: (x: T) => S,
+  f: (x: T, idx: number | CoordinateLike) => S,
```

Comment:
We know what type `idx` is (`number` or `CoordinateLike`) based on `shape`.  I think it's like this:

```
mapBoard(board, f: (x: T, idx: CoordinateLike) => S, shape: "2d");
mapBoard(board, f: (x: T, idx: number) => S, shape: "1d");
mapBoard(board, f: (x: T, idx: number) => S, shape: "flatten-2d-to-1d");
```

If TypeScript complains, no worries, but I think this would be most accurate!

Reply:
Yes typescript complains, the reason being that the function type `(x:T, idx: number) => S` is not a special case of `(x: T, idx: number | CoordinateLike) => S`.

I believe it would work if I change the type thusly:
`f: ((x: T, idx: CoordinateLike) => S) | ((x: T, idx: number) => S)`
... and use type assertion in the implementation function.

Reply:
> I believe it would work if I change the type thusly:
f: ((x: T, idx: CoordinateLike) => S) | ((x: T, idx: number) => S)

Oh nice!

> ... and use type assertion in the implementation function.

IMO, a little type fudging in implementation can be worth it for more accurate interface (but nothing to spend too much time on)

Reply:
I believe it would require changes where the functions are used, so I decided to land this regardless.